src/dashboard/infrastructure/content-types/content-types-collector.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong
+namespace Yoast\WP\SEO\Dashboard\Infrastructure\Content_Types;
+
+use Yoast\WP\SEO\Dashboard\Domain\Content_Types\Content_Type;
+use Yoast\WP\SEO\Helpers\Post_Type_Helper;
+/**
+ * Class that collects post types and relevant information.
+ */
+class Content_Types_Collector {
+
+	/**
+	 * The post type helper.
+	 *
+	 * @var Post_Type_Helper
+	 */
+	private $post_type_helper;
+
+	/**
+	 * The constructor.
+	 *
+	 * @param Post_Type_Helper $post_type_helper The post type helper.
+	 */
+	public function __construct(
+		Post_Type_Helper $post_type_helper
+	) {
+		$this->post_type_helper = $post_type_helper;
+	}
+
+	/**
+	 * Returns the content types array.
+	 *
+	 * @return array<string,Content_Type> The content types array.
+	 */
+	public function get_content_types(): array {
+		$content_types = [];
```

Comment:
Because I view this more as more a wrapper of the post type helper that we used to have in the repository.

In any case, I've made it using the list instead [here](https://github.com/Yoast/wordpress-seo/pull/21847/commits/188b4e49bdd169e4785f7024035d9ef8b918c5fd).